source/common/tracing/zipkin/span_context.cc

```diff
@@ -0,0 +1,122 @@
+#include "common/common/utility.h"
+
+#include "common/tracing/zipkin/zipkin_core_constants.h"
+#include "common/tracing/zipkin/span_context.h"
+
+namespace Zipkin {
+
+// String that separates the span-context fields in its string-serialized form.
+const std::string SpanContext::FIELD_SEPARATOR_ = ";";
+
+// String value corresponding to an empty span context.
+const std::string SpanContext::UNITIALIZED_SPAN_CONTEXT_ = "0000000000000000" + FIELD_SEPARATOR_ +
+                                                           "0000000000000000" + FIELD_SEPARATOR_ +
+                                                           "0000000000000000";
+
+// String with regular expression to match a 16-digit hexadecimal number.
+const std::string SpanContext::HEX_DIGIT_GROUP_REGEX_STR_ = "([0-9,a-z]{16})";
+
+// The static functions below are needed due to C++ inability to safely concatenate static strings
+// belonging to different compilation units (the initialization order is not guaranteed).
+
+const std::string& SpanContext::SPAN_CONTEXT_REGEX_STR() {
+  // ^([0-9,a-z]{16});([0-9,a-z]{16});([0-9,a-z]{16})((;(cs|sr|cr|ss))*)$
+  static std::string span_context_regex_str =
```

Comment:
This also has dangerous destruction ordering behavior if it leaks across compilation units, see https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables. Prefer:

```
static const std::string* span_context_regex_str = new std::string(...);
return *span_context_regex_str;
```